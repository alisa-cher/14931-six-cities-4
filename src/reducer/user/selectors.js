import NameSpace from "../name-space";

const getAuthorisationStatus = (state) => {
  return state[NameSpace.USER].authorizationStatus;
};

export {getAuthorisationStatus};
