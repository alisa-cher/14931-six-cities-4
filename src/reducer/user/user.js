import {Operation as DataOperation} from "../data/data";
import {mapHotels} from "../../data/adapter";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  userData: {}
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_USER: `SET_USER`
};

const ActionCreator = {
  setAuthStatus: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  setUser: (data) => {
    return {
      type: ActionType.SET_USER,
      payload: data,
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return Object.assign({}, state, {
        authorizationStatus: action.payload,
      });
    case ActionType.SET_USER:
      return Object.assign({}, state, {
        userData: action.payload,
      });
  }

  return state;
};

const Operation = {
  checkAuth: (adapter) => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.setAuthStatus(AuthorizationStatus.AUTH));
        const mappedData = adapter(response.data);
        dispatch(ActionCreator.setUser(mappedData));
        dispatch(DataOperation.getFavorites(mapHotels));
      })
      .catch(() => {
        dispatch(ActionCreator.setAuthStatus(AuthorizationStatus.NO_AUTH));
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then(() => {
        dispatch(ActionCreator.setAuthStatus(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        throw err;
      });
  },
};

export {reducer, ActionType, ActionCreator, AuthorizationStatus, Operation};
