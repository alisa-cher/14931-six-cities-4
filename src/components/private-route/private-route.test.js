import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route.jsx';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {MemoryRouter} from 'react-router';

const MockComponent = () => <div />;

it(`Should render private route component`, () => {
  const tree = renderer.create(
      <MemoryRouter initialEntries={[`/favorites`]}>
        <PrivateRoute
          exact
          path={`/favorites`}
          authorizationStatus={AuthorizationStatus.AUTH}
          render={() => <MockComponent />}
        />
      </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
