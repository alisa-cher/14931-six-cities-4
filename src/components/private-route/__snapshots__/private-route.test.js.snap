// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render private route component 1`] = `<div />`;
